geo: add encoding functionality to different formats

otan opened the pull request:
This commit adds encoding for KML, GeoJSON, WKT, WKB and WKBHex. Note EWKB
and EWKBHex is already provided by the identity function and String()
respectively. These will be plugged in at a later date.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/47549)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 7 of 7 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47549#-:-M5HpkZL3nsuxg2DvIm6:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/geo/encode.go, line 82 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47549#-M5Ho0yO5bw_ki5skemu:-M5Ho0yO5bw_ki5skemv:bwxqa00) ([raw file](https://github.com/cockroachdb/cockroach/blob/13b45580ac99c81cdf931088c6b68bebc5f67057/pkg/geo/encode.go#L82)):*
> ```Go
> 		return "", err
> 	}
> 	ret, err := wkbhex.Encode(t, ewkbEncodingFormat)
> ```

does WKBHex not include the SRID?
___
*[pkg/geo/parse.go, line 44 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47549#-M5Horr38T3XoTfBgS-a:-M5Horr38T3XoTfBgS-b:b-9n8ot7) ([raw file](https://github.com/cockroachdb/cockroach/blob/13b45580ac99c81cdf931088c6b68bebc5f67057/pkg/geo/parse.go#L44)):*
> ```Go
> 	}
> 
> 	// TODO(otan): parse GeoJSON by checking if character starts with '{'
> ```

which character? is this still a TODO in light of the `case {` above?


<!-- Sent from Reviewable.io -->

otan committed:
geo: add encoding functionality to different formats

This commit adds encoding for KML, GeoJSON, WKT, WKB and WKBHex. Note EWKB
and EWKBHex is already provided by the identity function and String()
respectively. These will be plugged in at a later date.

Release note: None

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47549)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @sumeerbhola)

___
*[pkg/geo/encode.go, line 82 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47549#-M5Ho0yO5bw_ki5skemu:-M5N1iWq7dfJ_pn9a48t:b-83og4a) ([raw file](https://github.com/cockroachdb/cockroach/blob/13b45580ac99c81cdf931088c6b68bebc5f67057/pkg/geo/encode.go#L82)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

does WKBHex not include the SRID?
</blockquote></details>

Nope!
___
*[pkg/geo/parse.go, line 44 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47549#-M5Horr38T3XoTfBgS-a:-M5N1xhfEaKikqW4oP5l:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/13b45580ac99c81cdf931088c6b68bebc5f67057/pkg/geo/parse.go#L44)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

which character? is this still a TODO in light of the `case {` above?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47549#-:-M5OF_ZG7odyf206il8l:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @sumeerbhola)



<!-- Sent from Reviewable.io -->

otan commented:
bors r=sumeerbhola

otan commented:
flake

bors r=sumeerbhola
